packages/wallet-service/src/api/txOutputs.ts

```diff
@@ -221,6 +225,24 @@ const _getFilteredTxOutputs = async (walletId: string, filters: IFilterTxOutput)
     }
   }
 
+  // Apply maxAmount filter if specified (returns UTXOs summing to at most maxAmount)
+  if (filters.maxAmount) {
+    let accumulatedAmount = 0n;
+    const selectedTxOutputs: DbTxOutput[] = [];
+
+    // txOutputs are sorted by value DESC from the database, so we iterate
+    // from smallest to largest to maximize the number of UTXOs within the limit
```

Comment:
Why do we want to "maximize" the number of utxos?
In the wallet-lib when an amount limit is given we usually try to select as few utxos as possible to avoid hitting the tx input limit (255)